ci-operator/step-registry/vsphere/deprovision/customized-resourcepool/vsphere-deprovision-customized-resourcepool-commands.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/govc.sh"
+unset SSL_CERT_FILE
+unset GOVC_TLS_CA_CERTS
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+
+rsp_config=$(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].topology.resourcePool')
+if [[ -z "$(govc pool.info -json ${rsp_config} | jq -r '.ResourcePools[].Vm[] | join(":")' | xargs govc ls -L | awk -F'/' '{print $NF}')" ]];then
```

Comment:
Resource pool specified in install-config is shared to all vSphere clusters, when you run this command, all machines created by CI jobs can be listed.
```
# govc pool.info -json /DEVQEdatacenter/host/DEVQEcluster/Resources/ipi-ci-clusters | jq -r '.ResourcePools[].Vm[] | join(":")' | xargs govc ls -L | awk -F'/' '{print $NF}'
ci-op-k83sn59i-182b3-h5sgz-worker-0-jwmqg
ci-op-k83sn59i-182b3-h5sgz-worker-0-56qdq
ci-op-k83sn59i-182b3-h5sgz-worker-0-9rp5q
ci-op-k83sn59i-182b3-h5sgz-master-1
ci-op-k83sn59i-182b3-h5sgz-master-2
ci-op-k83sn59i-182b3-h5sgz-master-0
ci-op-x08l6vxc-bab81-qprn2-worker-0-hcr8r
ci-op-x08l6vxc-bab81-qprn2-master-0
ci-op-x08l6vxc-bab81-qprn2-worker-0-tqw5t
ci-op-x08l6vxc-bab81-qprn2-master-2
ci-op-x08l6vxc-bab81-qprn2-master-1
ci-op-x08l6vxc-bab81-bastion
ci-op-x08l6vxc-bab81-qprn2-worker-0-bpsxb
```
At least you need to append filtering with `infraId` or `clusterName` + excluding bastion host to check if related machines are destroyed or not, else this step might be always failed.

Reply:
oh, you're right. will update.